### PR TITLE
Create Note component

### PR DESCRIPTION
**Pull request by colebemis**
## Problem

We often write "notes" in our documentation pages to call attention to specific content. However, we don't have a consistent way to format notes. Sometimes we _italicize_ notes. Sometimes we write notes __bold__. 

Additionally, none of these note formats do a great job of actually calling attention to the content of the note. 

## Solution

This PR adds a `Note` component that people can use to write notes in documentation pages. Here's an example of the `Note` component in action:

image

The `Note` component has three variants: info (default), warning, and danger:

image

👉 Check out the rest of the documentation for the Note component [here](https://doctocat-git-note-component.primer.vercel.app/doctocat/components/note).

Closes https://github.com/primer/doctocat/issues/110

### CODE REVIEW

File: theme/src/components/wrap-root-element.js [hunk 1]
```diff
@@ -32,6 +33,9 @@ const components = {
   ul: List,
   ol: List.withComponent('ol'),
   dl: DescriptionList,
+
+  // Shortcodes (https://mdxjs.com/blog/shortcodes)
+  Note,
```
Comment (colebemis) [inline theme/src/components/wrap-root-element.js:38]: Making `Note` a [shortcode](https://mdxjs.com/blog/shortcodes) allows people to use it in their markdown files without having to import it. I think this is a huge UX improvement. We should probably make `Caption` and `Do/Dont` shortcodes as well. Although, I intentionally didn't make those other components shortcodes in this PR to keep these changes focused on the `Note` component.

**Comment by ashygee** [comment]
I like the design of the `Note` component but I wonder if it needs to be as emphasized with the left border. 

Could it simpler just having a bg color and stroke? 
example of a note component with a light purple background and light purple stroke. all text is black.

**Comment by colebemis** [comment]
@ashygee Here's a comparison of the two styles:

| Option 1 | Option 2 |
| :-: | :-:|
| [🔗 Preview](https://doctocat-bipwne027.vercel.app/) | [🔗 Preview](https://doctocat-jevwz6sjo.vercel.app/) |
| Screen Shot 2020-08-26 at 12 41 25 PM | Screen Shot 2020-08-26 at 12 43 53 PM |
| Screen Shot 2020-08-26 at 12 41 15 PM  | Screen Shot 2020-08-26 at 12 46 06 PM |

Which do you prefer?

**Comment by ashygee** [comment]
> Which do you prefer?

@colebemis I actually prefer the simpler design. It doesn't distract as much but still feels *special*.

**Comment by colebemis** [comment]
@ashygee Sounds good to me. Since that style looks like our `Flash` component, I updated the code to actually use the `Flash` component from Primer React Components to implement `Note`. This brings up a couple of questions:

- Should this component still be called `Note` even though it's essentially the same as the `Flash` component?  

I'm leaning towards keeping it called `Note` because it has a different purpose than `Flash`. `Note` is used to draw attention to information in a document. `Flash` informs users of successful or pending actions. Renaming `Note` to `Flash` seems a little bit like exposing an implementation detail. Users don't care that `Note` is implemented with a `Flash` component, they just want to write a note in their document. I'd love to hear other people's thoughts on this (cc @emplums)

- `Flash` has a `success` variant. Should `Note` also have a `success` variant? In what contexts would we use a "success" note?

**Comment by ashygee** [comment]
> * Should this component still be called `Note` even though it's essentially the same as the `Flash` component?

I agree with your thinking and that it should still be called `Note` as it's functions is different from how we use the `Flash`. I +1 to keeping it as `Note` but I'll defer to @emplums and her recommendations as final say.

> * `Flash` has a `success` variant. Should `Note` also have a `success` variant? In what contexts would we use a "success" note?

I don't feel a `success` variant is necessary. The only instance is if we wanted to highlight a "You should do this!" tip but I think the Do/Don't component solves for this.

**Comment by emplums** [comment]
I personally prefer the first design with the thick left border over the flash design 😬 I feel like the side border attracts the eye more, vs the flash which I feel a bit conditioned to gloss over. I believe we went back and forth about this a few months ago though but I'm not sure what we settled on.

I also wonder if we would normally encourage folks to use a Flash inside of a body of text? If not, would using it in the docs encourage a pattern we don't really want used in .com?

**Comment by colebemis** [comment]
FWIW I also prefer the left border design. I think it makes sense for `Note` and `Flash `to visually look different because they have different purposes. However, I'm happy to defer to @ashygee's judgment on this.

Regardless of which design we go with, I think we're all in agreement that this component should still be called `Note` and not `Flash`.

**Comment by ashygee** [comment]
@emplums I agree that it draws your eye more. We can go with this design. My original line of thinking was to use something that matched with what we have in Primer and was worried that this might feel too closely related to the `Toast` but it makes sense that we would want readers to focus on these items. Also, it does make sense to feel less related to a `Flash`.

File: docs/content/components/note.mdx [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+title: Note
+status: New
+source: https://github.com/primer/doctocat/blob/master/theme/src/components/note.js
+---
+
+Use the `Note` component to call out information in documentation pages.
+
+## Usage
+
+`Note` is a [shortcode](https://mdxjs.com/blog/shortcodes), meaning it's globally available in all `.md` and `.mdx` files. So you can use the `Note` component in any `.md` or `.mdx` file without importing it.
```
Comment (emplums) [inline docs/content/components/note.mdx:11]: TIL about shortcodes! Are all our other documentation components set up as shortcodes?

**Pull request merged**